pkg/apis/options/providers.go

```diff
@@ -0,0 +1,172 @@
+
+type BitbucketOptions struct {
+	// BitbucketTeam sets restrict logins to members of this team
+	BitbucketTeam string `json:"bitbucketTeam,omitempty"`
+	// BitbucketRepository sets restrict logins to user with access to this repository
+	BitbucketRepository string `json:"bitbucketRepository,omitempty"`
+}
+
+type GitHubOptions struct {
+	// GitHubOrg sets restrict logins to members of this organisation
+	GitHubOrg string `json:"githubOrg,omitempty"`
+	// GitHubTeam sets restrict logins to members of this team
+	GitHubTeam string `json:"githubTeam,omitempty"`
+	// GitHubRepo sets restrict logins to collaborators of this repository
+	GitHubRepo string `json:"githubRepo,omitempty"`
+	// GitHubToken is the token to use when verifying repository collaborators
+	// it must have push access to the repository
+	GitHubToken string `json:"githubToken,omitempty"`
+	// GitHubUsers allows users with these usernames to login
+	// even if they do not belong to the specified org and team or collaborators
+	GitHubUsers []string `json:"githubUsers,omitempty"`
+}
+
+type GitLabOptions struct {
+	// GitLabGroup sets restrict logins to members of this group
+	GitLabGroup []string `json:"gitlabGroups,omitempty"`
+}
+
+type GoogleOptions struct {
+	// GoogleGroups sets restrict logins to members of this google group
+	GoogleGroups []string `json:"googleGroup,omitempty"`
+	// GoogleAdminEmail is the google admin to impersonate for api calls
+	GoogleAdminEmail string `json:"googleAdminEmail,omitempty"`
+	// GoogleServiceAccountJSON is the path to the service account json credentials
+	GoogleServiceAccountJSON string `json:"googleServiceAccountJson,omitempty"`
+}
+
+type OIDCOptions struct {
+	// OIDCIssuerURL is the OpenID Connect issuer URL
+	// ie: https://accounts.google.com
+	OIDCIssuerURL string `json:"oidcIssuerURL,omitempty"`
+	// InsecureOIDCAllowUnverifiedEmail prevents failures if an email address in an id_token is not verified
+	// default set to 'false'
+	InsecureOIDCAllowUnverifiedEmail bool `json:"insecureOidcAllowUnverifiedEmail,omitempty"`
+	// InsecureOIDCSkipIssuerVerification skips verification of ID token issuers. When false, ID Token Issuers must match the OIDC discovery URL
+	// default set to 'false'
+	InsecureOIDCSkipIssuerVerification bool `json:"insecureOidcSkipIssuerVerification,omitempty"`
+	// SkipOIDCDiscovery allows to skip OIDC discovery and use manually supplied Endpoints
+	// default set to 'false'
+	SkipOIDCDiscovery bool `json:"skipOidcDiscovery,omitempty"`
+	// OIDCJwksURL is the OpenID Connect JWKS URL
+	// ie: https://www.googleapis.com/oauth2/v3/certs
+	OIDCJwksURL string `json:"oidcJwksURL,omitempty"`
+	// OIDCGroupsClaim indicates which claim contains the user groups
+	// default set to 'groups'
+	OIDCGroupsClaim string `json:"oidcGroupsClaim,omitempty"`
+	// UserIDClaim indicates which claim contains the user ID
+	// default set to 'email'
+	UserIDClaim string `json:"userIDClaim,omitempty"`
+}
+
+type LoginGovOptions struct {
```

Comment:
FYI - This is another provider that is OIDC based and borrows some of the OIDC options already I believe.

Reply:
didn't see anything that needed to be addresses while rebasing, hope I am not missing anything here.